### PR TITLE
fix(common): Mark `\r` as a line ending

### PR DESCRIPTION
**Pull request by kdy1**
**Description:**

**Related issue:**

 - Closes https://github.com/swc-project/swc/issues/6694.

### CODE REVIEW

**Review by kdy1 — COMMENTED** [review]
swc-bump:
 - swc_common

File: crates/swc_common/src/syntax_pos/analyze_source_file.rs [hunk 1]
```diff
@@ -47,167 +48,6 @@ pub fn analyze_source_file(
     (lines, multi_byte_chars, non_narrow_chars)
 }
 
-cfg_if::cfg_if! {
```
Comment (kdy1) [inline crates/swc_common/src/syntax_pos/analyze_source_file.rs:50]: This is not ideal, but I'm not sure if this optimization is really required.

File: crates/swc_common/src/syntax_pos/analyze_source_file.rs [hunk 2]
```diff
@@ -240,7 +80,7 @@ fn analyze_source_file_generic(
             let pos = BytePos::from_usize(i) + output_offset;
 
             match byte {
-                b'\n' => {
+                b'\n' | b'\r' => {
```
Comment (jridgewell) [inline crates/swc_common/src/syntax_pos/analyze_source_file.rs:83]: I think we need to handle `\r\n` as a single line ending and not two line endings (unless that's been normalized as some prior point I'm not seeing).

**Review by jridgewell — CHANGES_REQUESTED** [review]

File: crates/swc_common/src/syntax_pos/analyze_source_file.rs (hunk 2 above)
Comment (kdy1) [reply crates/swc_common/src/syntax_pos/analyze_source_file.rs]: Did it, and I'll add some tests for it.

File: crates/swc_common/src/syntax_pos/analyze_source_file.rs [hunk 3]
```diff
@@ -240,6 +80,14 @@ fn analyze_source_file_generic(
             let pos = BytePos::from_usize(i) + output_offset;
 
             match byte {
+                b'\r' => {
+                    lines.push(pos + BytePos(1));
```
Comment (jridgewell) [inline crates/swc_common/src/syntax_pos/analyze_source_file.rs:84]: Won't this need to be `+ BytePos(2)` if we're hitting a `\r\n`?

File: crates/swc/tests/source_map.rs [hunk 4]
```diff
@@ -424,3 +426,77 @@ fn issue_4578() {
...
+            Ok(result) => {
+                println!("{}", result.code);
+
+                assert!(result.map.is_some());
+                let map = result.map.unwrap();
+
+                let source_map = sourcemap::SourceMap::from_slice(map.as_bytes())
+                    .expect("failed to deserialize sourcemap");
+
+                dbg!(&source_map);
```
Comment (jridgewell) [inline crates/swc/tests/source_map.rs:485]: Are the debug and prints supposed to still be here?

**Review by jridgewell — CHANGES_REQUESTED** [review]

**Review by jridgewell — DISMISSED** [review]

**Pull request merged**